### ISSUE TITLE
Bug: AdditionalProperties set to true leads to Nil Pointer

### ISSUE THREAD

**Issue by floric** [Skarlso/crd-to-sample-yaml#146]
Hey :)

Thanks for the tool. We added `additionalProperties: true` today for one spec. This lead to a failing HTML report with the following error:

```
$ cty generate crd -r ${CENTRAL_DIR} -o ${CI_PROJECT_DIR}/public/index.html -f html --no-random
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x168 pc=0x9e1de7]
goroutine 1 [running]:
github.com/Skarlso/crd-to-sample-yaml/pkg.parseCRD(0xc0005[27](xxxx#L27)6e0, {0xc0004cd9e0, 0x8}, 0x0, {0xc0004cd950, 0xd}, {0xc0004cd970, 0xe}, {0x18ceb00, 0x4, ...}, ...)
	github.com/Skarlso/crd-to-sample-yaml/pkg/create_html_output.go:[28](xxx#L28)5 +0xda7
github.com/Skarlso/crd-to-sample-yaml/pkg.generate({0xc0004cd9e0, 0x8}, {0xc0004cd950, 0xd}, {0xc0004cd970, 0xe}, 0xc0005e0e00, 0x1?, 0xc00054f868)
	github.com/Skarlso/crd-to-sample-yaml/pkg/create_html_output.go:167 +0xf0
github.com/Skarlso/crd-to-sample-yaml/pkg.RenderContent({0x1159108, 0xc00024a0c8}, {0xc00045c000, 0x20, 0xc0002b3a98?}, {0x0?, 0x0?, 0x0?})
	github.com/Skarlso/crd-to-sample-yaml/pkg/create_html_output.go:106 +0xb85
github.com/Skarlso/crd-to-sample-yaml/cmd.runGenerate(0xc0000c0f00?, {0xf52fb2?, 0x4?, 0xf52fb6?})
	github.com/Skarlso/crd-to-sample-yaml/cmd/crd.go:101 +0x2a8
github.com/spf13/cobra.(*Command).execute(0x18d94a0, {0xc0004dee00, 0x7, 0x7})
	github.com/spf13/cobra@v1.8.1/command.go:985 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0x18d9780)
	github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/Skarlso/crd-to-sample-yaml/cmd.Execute(...)
	github.com/Skarlso/crd-to-sample-yaml/cmd/root.go:18
main.main()
	github.com/Skarlso/crd-to-sample-yaml/main.go:19 +0x5b
```

This is caused by the assumption, that `additionalProperties` is always a map. In our case we set it to `true` which is also valid to the Kubernetes API. So we will need a check, if the value is true here:

https://github.com/Skarlso/crd-to-sample-yaml/blob/main/pkg/create_html_output.go#L282

Am I right or does it work in a different way? Kind regards, Florian

**Comment by Skarlso** [comment]
Ah thanks very much for reporting this!

**Comment by Skarlso** [comment]
Yes you're right but also in a couple of other places. 😁

**Comment by Skarlso** [comment]
Hm, can you give me the CRD please? It's not the additional property. Or it just doesn't have a schema or something.

**Comment by Skarlso** [comment]
Ah I see I added that check elsewhere but didn't add that check in the html generator.. :)

**Comment by Skarlso** [comment]
Can you try with v1.0.5 please? :)

**Comment by floric** [comment]
> Can you try with v1.0.5 please? :)

wow, thanks a lot :) We will give it a try on Monday.

**Comment by floric** [comment]
Hey :) I just tested it. The exectution succeeds without the nil pointer. But the result for the spec with `additionalProperties` is now empty.

Spec:
```yaml
 spec:
  additionalProperties: true
  x-kubernetes-preserve-unknown-fields: true
  description: some
  properties:
    secretKey:
      example: token
      pattern: ^[0-9a-zA-Z \@\*\_\-.\,/:]*$
      type: string
    secretName:
      description: secret name
      pattern: ^[0-9a-zA-Z \@\*\_\-.\,/:]*$
      type: string
  required:
    - secretKey
    - secretName
  type: object
```

The corresponding property in the HTML report contains only the description, but no fields (secretKey, secretName). The example still contains both fields in contrast.

Does it make sense to still render the base required properties?

**Comment by Skarlso** [comment]
Ah. Interesting. Let me check that. Can you paste the whole thing please so I can write a unit test? Thanks

**Comment by floric** [comment]
> Ah. Interesting. Let me check that. Can you paste the whole thing please so I can write a unit test? Thanks

Thanks for the super fast reply :D

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: abc
spec:
  group: abc.dp.db.de
  names:
    kind: ProviderConfig
    listKind: ProviderConfigList
    plural: providerconfigs
    singular: providerconfig
  scope: Namespaced
  versions:
    - deprecated: false
      name: v1alpha1
      schema:
        openAPIV3Schema:
          properties:
            apiVersion:
              enum:
                - abc.dp.db.de/v1alpha1
              pattern: ^[0-9a-zA-Z \@\*\_\-.\,/:]*$
              type: string
            kind:
              enum:
                - ProviderConfig
              pattern: ^[0-9a-zA-Z \@\*\_\-.\,/:]*$
              type: string
            metadata:
              properties:
                name:
                  pattern: ^[0-9a-zA-Z \@\*\_\-.\,/:]*$
                  type: string
                namespace:
                  pattern: ^[0-9a-zA-Z \@\*\_\-.\,/:]*$
                  type: string
              required:
                - name
              type: object
            spec:
              additionalProperties: true
              description: abc
              properties:
                secretKey:
                  description: abc
                  example: token
                  pattern: ^[0-9a-zA-Z \@\*\_\-.\,/:]*$
                  type: string
                secretName:
                  description: abc
                  pattern: ^[0-9a-zA-Z \@\*\_\-.\,/:]*$
                  type: string
              required:
                - secretKey
                - secretName
              type: object
          required:
            - spec
            - apiVersion
            - kind
            - metadata
          type: object
      served: true
      storage: true
```

**Comment by Skarlso** [comment]
ah, found the bugger... will have a fix shortyl

**Comment by Skarlso** [comment]
Actually, the website had the same problem hah.

**Comment by Skarlso** [comment]
Okay, the fix should be released [v1.0.6](https://github.com/Skarlso/crd-to-sample-yaml/releases/tag/v1.0.6).

**Comment by floric** [comment]
Great! :) It's fixed now. Great work again.
